### PR TITLE
Disable sage fields

### PR DESCRIPTION
**Pull request by Hrishabh17**

### CODE REVIEW

File: apps/sage300/helpers.py [hunk 1]
```diff
@@ -62,3 +67,76 @@ def sync_dimensions(sage300_credential: Sage300Credential, workspace_id: int) ->
...
+        'value__in': [projects_map['value'] for projects_map in projects_to_disable.values()]
+    }
+
+    value_id_map = {v['value']: k for k, v in projects_to_disable.items()}
+
+    expense_attributes = ExpenseAttribute.objects.filter(**filters)
+
+    bulk_payload = []
+    for expense_attribute in expense_attributes:
+        code = value_id_map.get(expense_attribute.value, None)
```
Comment (ashwin1111) [inline apps/sage300/helpers.py:89]: better variable name pls

File: apps/sage300/helpers.py [hunk 2]
```diff
@@ -62,3 +67,76 @@ def sync_dimensions(sage300_credential: Sage300Credential, workspace_id: int) ->
         except Exception as exception:
             # Log any exceptions that occur during synchronization
             logger.info(exception)
+
+
+def disable_projects(workspace_id: int, projects_to_disable: Dict):
```
Comment (ashwin1111) [inline apps/sage300/helpers.py:72]: just thinking if we could put this code in fyle_integration_imports. For sage300, it's not used, but thinking if we can do this for other apps

File: apps/sage300/helpers.py [hunk 3]
```diff
@@ -62,3 +67,76 @@ def sync_dimensions(sage300_credential: Sage300Credential, workspace_id: int) ->
...
+    sync_after = datetime.now(timezone.utc)
+
+    fyle_credentials = FyleCredential.objects.get(workspace_id=workspace_id)
+    platform = PlatformConnector(fyle_credentials=fyle_credentials)
+
+    if bulk_payload:
+        platform.projects.post_bulk(bulk_payload)
+        platform.projects.sync(sync_after=sync_after)
+
+    update_and_disable_cost_code(workspace_id, projects_to_disable, platform)
```
Comment (ashwin1111) [inline apps/sage300/helpers.py:113]: check if deps is enabled and call this / handle this scenario in that func

File: apps/sage300/helpers.py [hunk 4]
```diff
@@ -62,3 +67,76 @@ def sync_dimensions(sage300_credential: Sage300Credential, workspace_id: int) ->
...
+    Update the job_name in CostCategory and disable the old cost code in Fyle
+    """
+    dependent_field_setting = DependentFieldSetting.objects.filter(workspace_id=workspace_id).first()
+
+    filters = {
+        'job_id__in':list(cost_codes_to_disable.keys()),
+        'workspace_id': workspace_id
+    }
+
+    post_dependent_cost_code(dependent_field_setting, platform, filters, is_enabled=False)
```
Comment (ashwin1111) [inline apps/sage300/helpers.py:127]: why are we calling this?

Comment (ashwin1111) [inline apps/sage300/helpers.py:127]: can we call this func after bulk update is done?

Comment (ashwin1111) [reply apps/sage300/helpers.py:127]: what if post() failed, there is no way we could re try that action

File: apps/sage300/helpers.py [hunk 5]
```diff
@@ -62,3 +67,96 @@ def sync_dimensions(sage300_credential: Sage300Credential, workspace_id: int) ->
...
+    platform = PlatformConnector(fyle_credentials=fyle_credentials)
+
+    filters = {
+        'workspace_id': workspace_id,
+        'attribute_type': 'PROJECT',
+        'value__in': [projects_map['value'] for projects_map in projects_to_disable.values()]
+    }
+
+    # Expense attribute value map is as follows: {old_project_name: destination_id}
+    expense_attribute_value_map = {v['value']: k for k, v in projects_to_disable.items()}
```
Comment (ruuushhh) [inline apps/sage300/helpers.py:95]: can we do better naming here

**Review by ruuushhh — APPROVED** [review]

File: tests/test_sage300/test_helpers.py [hunk 6]
```diff
@@ -83,4 +88,95 @@ def test():
...
+    disable_projects(workspace_id, projects_to_disable)
+
+    projects_to_disable = {
+        'destination_id': {
+            'value': 'old_project_2',
+            'updated_value': 'new_project'
+        }
+    }
+
+    disable_projects(workspace_id, projects_to_disable)
```
Comment (ashwin1111) [inline tests/test_sage300/test_helpers.py:131]: why are we not asserting anything in this test?

Comment (Hrishabh17) [reply tests/test_sage300/test_helpers.py:131]: added assertions

**Pull request merged**